### PR TITLE
add token transfer support to execute msg, custom fees to contract constructor

### PR DESCRIPTION
**Pull request by udit-gulati**

### CODE REVIEW

File: packages/polar/src/lib/client.ts [hunk 1]
```diff
@@ -1,28 +1,29 @@
 import { info } from "console";
-import { CosmWasmClient, EnigmaUtils, ExecuteResult, Secp256k1Pen, SigningCosmWasmClient } from "secretjs";
+import { CosmWasmClient, EnigmaUtils, ExecuteResult, FeeTable, Secp256k1Pen, SigningCosmWasmClient } from "secretjs";
 
 import { Account, Network } from "../types";
 
 export function getClient (network: Network): CosmWasmClient {
-  info(`Creating client for network: ${network.name}`);
+  // info(`Creating client for network: ${network.name}`);  no need to log this
```
Comment (coco-sha) [inline packages/polar/src/lib/client.ts:7]: could you please remove it ?

File: packages/polar/src/lib/client.ts [hunk 2]
```diff
@@ -1,28 +1,29 @@
 import { info } from "console";
-import { CosmWasmClient, EnigmaUtils, ExecuteResult, Secp256k1Pen, SigningCosmWasmClient } from "secretjs";
```
Comment (coco-sha) [inline packages/polar/src/lib/client.ts:2]: linter error in above line.

File: packages/polar/src/lib/client.ts [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
 }
 
 export async function getSigningClient (
   network: Network,
-  account: Account
+  account: Account,
+  customFees?: Partial<FeeTable> | undefined
 ): Promise<SigningCosmWasmClient> {
-  info(`Creating signing client for network: ${network.name}`);
+  // info(`Creating signing client for network: ${network.name}`);  no need to log this
```
Comment (coco-sha) [inline packages/polar/src/lib/client.ts:18]: please remove it.

File: packages/polar/sample-project/scripts/sample-script.js [hunk 4]
```diff
@@ -1,8 +1,19 @@
 const { Contract, getAccountByName } = require("secret-polar");
 
 async function run (runtimeEnv) {
+  const customFees = {
```
Comment (coco-sha) [inline packages/polar/sample-project/scripts/sample-script.js:4]: this should be a part of config file

File: packages/polar/src/lib/client.ts [hunk 5]
```diff
@@ -1,28 +1,29 @@
...
   const signingPen = await Secp256k1Pen.fromMnemonic(account.mnemonic);
   const txEncryptionSeed = EnigmaUtils.GenerateNewSeed();
   return new SigningCosmWasmClient(
     network.config.endpoint,
     account.address,
     (signBytes) => signingPen.sign(signBytes),
     network.config.seed ?? txEncryptionSeed,
-    network.config.fees,
+    customFees,
```
Comment (coco-sha) [inline packages/polar/src/lib/client.ts:26]: this should be fetched from config file

File: packages/polar/src/lib/deploy/contract.ts [hunk 6]
```diff
@@ -109,8 +108,9 @@ export class Contract {
     [name: string]: ContractFunction<any> // eslint-disable-line  @typescript-eslint/no-explicit-any
   };
 
-  constructor (contractName: string) {
+  constructor (contractName: string, customFees?: Partial<FeeTable>) {
```
Comment (coco-sha) [inline packages/polar/src/lib/deploy/contract.ts:111]: customFee should be fetched from config file and type checking should be done. type FeeTable will be used to typecheck inputs

File: packages/polar/src/lib/deploy/contract.ts [hunk 7]
```diff
@@ -287,15 +288,17 @@ export class Contract {
...
 
     const msgData: { [key: string]: object } = {}; // eslint-disable-line @typescript-eslint/ban-types
     msgData[methodName] = callArgs;
     console.log(this.contractAddress, msgData);
     // Send the same handleMsg to increment multiple times
     return await signingClient.execute(
       this.contractAddress,
-      msgData
+      msgData,
+      undefined,
```
Comment (coco-sha) [inline packages/polar/src/lib/deploy/contract.ts:300]: what is the use of this variable ? do we need undefined here ?

Comment (udit-gulati) [reply packages/polar/src/lib/deploy/contract.ts:300]: I needed to skip this optional parameter `memo`, passing it as `undefined` is the only way I found.

File: packages/polar/src/lib/deploy/contract.ts [hunk 8]
```diff
@@ -54,32 +56,27 @@ function buildSend (
   argNames: AbiParam[]
 ): ContractFunction<any> { // eslint-disable-line  @typescript-eslint/no-explicit-any
   return async function (
+    account: Account | UserAccount,
+    transferAmount: readonly Coin[] | undefined,
```
Comment (coco-sha) [inline packages/polar/src/lib/deploy/contract.ts:60]:
```suggestion
    transferAmount?: readonly Coin[],
```

**Review by coco-sha — APPROVED** [review]

**Pull request merged**